independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/CreateExtension.java

```diff
@@ -355,7 +355,7 @@ public CreateExtensionCommandHandler prepare() throws QuarkusCommandException {
         final Optional<String> quarkusVersion = data.getStringValue(QUARKUS_VERSION);
         // in 2.10.0.CR1 quarkus-bootstrap-maven-plugin was deprecated in favor of quarkus-extension-maven-plugin
         if (quarkusVersion.isPresent() &&
-                new DefaultArtifactVersion("2.10.0.CR1").compareTo(new DefaultArtifactVersion(quarkusVersion.get())) > 0) {
+                new ComparableVersion("2.10.0.CR1").compareTo(new ComparableVersion(quarkusVersion.get())) > 0) {
```

Comment:
`DefaultArtifactVersion` uses `ComparableVersion` internally, so I guess this doesn't change anything

Reply:
I wouldn't be that stupid :).

`ComparableVersion` is fine, I tested it and I added tests there: https://github.com/quarkusio/conversational-release-action/blob/main/src/test/java/io/quarkus/bot/release/VersionSchemeTest.java#L28 to check things would be fine.

What's not working is the code in `DefaultArtifactVersion` that extracts major/minor/... (and that's what I missed). It's a different path and this one is broken.
`DefaultArtifactVersion` sorting is actually fine as based on `ComparableVersion` so the only thing that was actually wrong is the code that was extracting major/minor.

So actually using `ComparableVersion` is fine. We could switch to SmallRye Common Version though, I don't mind.

Reply:
And to clarify, the reason I changed it wasn't to fix it (except for the extraction): it was to avoid the additional layer that was useless... and potentially harmful if we start using other methods of `DefaultArtifactVersion`.